### PR TITLE
Update to ISAL_2.21

### PR DESCRIPTION
**Pull request by pnvaidya**
Updated compression to use ISAL-2.21 release. Used ISAL now for level 1 and level 2 compression.

### CODE REVIEW

File: src/main/native/compression/CMakeLists.txt [hunk 1]
```diff
@@ -45,6 +45,11 @@ set_target_properties(isal PROPERTIES IMPORTED_LOCATION
   ${CMAKE_CURRENT_BINARY_DIR}/isalExternal-prefix/lib/libisal.a
 )
 
+#---------------------------------------------------------------------
+# common includes
+#---------------------------------------------------------------------
+include_directories(../common)
+
```
Comment (gspowley) [inline src/main/native/compression/CMakeLists.txt]: Is adding a new `include_directories` required?

File: src/main/native/compression/IntelDeflater.cc [hunk 2]
```diff
@@ -28,6 +28,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include "avx.h"
```
Comment (gspowley) [inline src/main/native/compression/IntelDeflater.cc]: Is this include required?

File: src/test/java/com/intel/gkl/compression/DeflaterIntegrationTest.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public Deflater makeDeflater(final int compressionLevel, final boolean nowrap) {
         log.info("deflater level, time (sec), filesize");
 
         for (DeflaterFactory deflaterFactory : deflaterFactories) {
-            for (int compressionLevel = 0; compressionLevel < 10; compressionLevel++) {
+            for (int compressionLevel = 1; compressionLevel < 4; compressionLevel++) {
```
Comment (gspowley) [inline src/test/java/com/intel/gkl/compression/DeflaterIntegrationTest.java]: This line should not change.

**Comment by pnvaidya** [comment]
Uses the latest ISAL 2.21 for Level 1 and Level 2 compression.

**Pull request merged**